app/models/vote.rb

```diff
@@ -0,0 +1,16 @@
+class Vote < ApplicationRecord
+  belongs_to :voter,
+             class_name:  :User,
+             foreign_key: :user_id
+
+  belongs_to :voteable,
+             polymorphic: true
+
+  validates :user_id,
+            uniqueness: { scope: [:voteable_id, :voteable_type] }
+
+  validates :value,
+            presence: true,
+            numericality: true,
+            inclusion: { in: [1, -1] }
```

Comment:
this is super neat, but I'm curious whether the `numericality: true` is adding any value here on top of the `inclusion` check 🤔

Reply:
Only in a circular way - a test for numericality was failing when `numericality: true` was absent. When I was first writing the tests, I started with deciding that `value` needed to be a number. I prefer to strictly allow only 1 or negative 1. I'm going to take out the numericality test and this line.